### PR TITLE
[#66] Reconcile streaming processor

### PR DESCRIPTION
**Pull request by fbiville**

### CODE REVIEW

File: pkg/controllers/streaming/processor_controller.go [hunk 1]
```diff
@@ -18,36 +18,225 @@ package streaming
 
 import (
 	"context"
-
+	"fmt"
 	"github.com/go-logr/logr"
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: nit: restore newline

File: pkg/controllers/streaming/processor_controller.go [hunk 2]
```diff
@@ -18,36 +18,225 @@ package streaming
...
+		if updateErr := r.Status().Update(ctx, processor); updateErr != nil {
+			log.Error(updateErr, "unable to update Processor status", "processor", processor)
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+	}
+	return result, err
 }
 
 func (r *ProcessorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueTrackedResources := &handler.EnqueueRequestsFromMapFunc{
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: for consistency with other reconcilers, move `SetupWithManager` to the bottom of the file

File: pkg/controllers/streaming/processor_controller.go [hunk 3]
```diff
@@ -18,36 +18,225 @@ package streaming
...
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Tracker tracker.Tracker
 }
 
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: add missing RBAC policy for Functions and Streams `get;list;watch`

File: pkg/controllers/streaming/processor_controller.go [hunk 4]
```diff
@@ -18,36 +18,225 @@ package streaming
...
+
+	// We may be reading a version of the object that was stored at an older version
+	// and may not have had all of the assumed defaults specified.  This won't result
+	// in this getting written back to the API Server, but lets downstream logic make
+	// assumptions about defaulting.
+	processor.SetDefaults(ctx)
+
+	processor.Status.InitializeConditions()
+
+	// resolve function
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: pull into a separate method, similar to https://github.com/projectriff/system/blob/87f386572f252b7a04852cdf97f50fd6188b4547/pkg/controllers/core/deployer_controller.go#L127

File: pkg/controllers/streaming/processor_controller.go [hunk 5]
```diff
@@ -18,36 +18,225 @@ package streaming
...
+				requests = append(requests, reconcile.Request{NamespacedName: item})
+			}
+			return requests
+		}),
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&streamingv1alpha1.Processor{}).
+		Owns(&appsv1.Deployment{}).
+		Watches(&source.Kind{Type: &buildv1alpha1.Function{}}, enqueueTrackedResources).
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: Streams are being watched as well

File: pkg/controllers/streaming/processor_controller.go [hunk 6]
```diff
@@ -18,36 +18,225 @@ package streaming
...
+	// We may be reading a version of the object that was stored at an older version
+	// and may not have had all of the assumed defaults specified.  This won't result
+	// in this getting written back to the API Server, but lets downstream logic make
+	// assumptions about defaulting.
+	processor.SetDefaults(ctx)
+
+	processor.Status.InitializeConditions()
+
+	// resolve function
+	functionName := types.NamespacedName{
```
Comment (scothis) [inline pkg/controllers/streaming/processor_controller.go]: Name != NamespacedName

File: pkg/apis/streaming/v1alpha1/resources/deployment.go [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"app": proc.Name,
+					},
+				},
+				Spec: corev1.PodSpec{
+					Containers: []corev1.Container{
+						{
+							Name:            "processor",
+							Image:           "ericbottard/processor:grpc",
```
[the remark this answers is not in the document]
Comment (ericbottard) [reply pkg/apis/streaming/v1alpha1/resources/deployment.go]: See recipe in this commit https://github.com/projectriff/system/pull/76/commits/fe34dcf65758966ad351735bbc6cf23c4888c332

**Comment by ericbottard** [comment]
Superseded by https://github.com/projectriff/system/pull/85.
Sorry for the re-opening, but pushing to Florent's master seemed to have strange side effects.

**Pull request closed without merging**